File: node-client-sdk/src/sdk/models/operations/get.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";


export class GetQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=foo" })
  foo: string;
}


export class GetRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: GetQueryParams;
}


export class Get200ApplicationJson extends SpeakeasyBase {
  @Metadata({ data: "json, name=args" })
  args?: Map<string, unknown>;

  @Metadata({ data: "json, name=headers" })
  headers?: Map<string, unknown>;

  @Metadata({ data: "json, name=origin" })
  origin?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class GetResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  get200ApplicationJsonObject?: Get200ApplicationJson;
}

File: go-client-sdk/pkg/models/operations/get.go
package operations

type GetQueryParams struct {
	Foo string `queryParam:"style=form,explode=true,name=foo"`
}

type GetRequest struct {
	QueryParams GetQueryParams
}

type Get200ApplicationJSON struct {
	Args    map[string]interface{} `json:"args,omitempty"`
	Headers map[string]interface{} `json:"headers,omitempty"`
	Origin  *string                `json:"origin,omitempty"`
	URL     *string                `json:"url,omitempty"`
}

type GetResponse struct {
	ContentType                 string
	StatusCode                  int64
	Get200ApplicationJSONObject *Get200ApplicationJSON
}
